core/src/main/java/org/infinispan/factories/InboundInvocationHandlerFactory.java

```diff
@@ -22,6 +23,9 @@
             configuration.transaction().transactionProtocol().isTotalOrder()) {
          //noinspection unchecked
          return (T) new TotalOrderTxPerCacheInboundInvocationHandler();
+      } else if (configuration.transaction().transactionMode().isTransactional()){
```

Comment:
Just have put another if inside of the else if.  Seems a little more clear as I was first confused as to why we were calling isTransactional twice.